### ISSUE TITLE
Nyxt 2.0.0 Built on Mac OS Mojave but will not run

### ISSUE THREAD

**Issue by ln0a** [atlas-engineer/nyxt#1066]
As the title suggests, Nyxt 2.0.0 compiles on Mojave but with `WARNINGS`. When running the Nyxt.app binary the program opens but the window does not open. When running the binary from the terminal it spits out the same `WARNINGS` from compilation.

--
objc[92065]: Class GNotificationCenterDelegate is implemented in both /usr/local/lib/libgio-2.0.dylib (0x4408a58) and /opt/local/lib/libgio-2.0.0.dylib (0x8bd0498). One of the two will be used. Which one is undefined.
objc[92065]: Class GdkQuartzView is implemented in both /usr/local/Cellar/gtk+3/3.24.24/lib/libgdk-3.0.dylib (0x50b7ec8) and /opt/local/lib/libgdk-3.0.dylib (0x88f54e8). One of the two will be used. Which one is undefined.
objc[92065]: Class GdkQuartzNSWindow is implemented in both /usr/local/Cellar/gtk+3/3.24.24/lib/libgdk-3.0.dylib (0x50b7f18) and /opt/local/lib/libgdk-3.0.dylib (0x88f5538). One of the two will be used. Which one is undefined.
objc[92065]: Class gdkCoreCursor is implemented in both /usr/local/Cellar/gtk+3/3.24.24/lib/libgdk-3.0.dylib (0x50b7f68) and /opt/local/lib/libgdk-3.0.dylib (0x88f5588). One of the two will be used. Which one is undefined.
objc[92065]: Class GtkQuartzStatusIcon is implemented in both /usr/local/lib/libgtk-3.0.dylib (0x4f256b8) and /opt/local/lib/libgtk-3.0.dylib (0x8764718). One of the two will be used. Which one is undefined.
objc[92065]: Class GtkClipboardOwner is implemented in both /usr/local/lib/libgtk-3.0.dylib (0x4f25618) and /opt/local/lib/libgtk-3.0.dylib (0x8764768). One of the two will be used. Which one is undefined.
objc[92065]: Class GtkDragSourceOwner is implemented in both /usr/local/lib/libgtk-3.0.dylib (0x4f25668) and /opt/local/lib/libgtk-3.0.dylib (0x87647b8). One of the two will be used. Which one is undefined.
objc[92065]: Class ResultReceiver is implemented in both /usr/local/lib/libgtk-3.0.dylib (0x4f25730) and /opt/local/lib/libgtk-3.0.dylib (0x8764830). One of the two will be used. Which one is undefined.
objc[92065]: Class GtkApplicationQuartzDelegate is implemented in both /usr/local/lib/libgtk-3.0.dylib (0x4f25758) and /opt/local/lib/libgtk-3.0.dylib (0x8764858). One of the two will be used. Which one is undefined.
objc[92065]: Class GNSMenuItem is implemented in both /usr/local/lib/libgtk-3.0.dylib (0x4f257a8) and /opt/local/lib/libgtk-3.0.dylib (0x87648a8). One of the two will be used. Which one is undefined.
objc[92065]: Class GNSMenu is implemented in both /usr/local/lib/libgtk-3.0.dylib (0x4f257d0) and /opt/local/lib/libgtk-3.0.dylib (0x87648d0). One of the two will be used. Which one is undefined.
objc[92065]: Class FilterComboBox is implemented in both /usr/local/lib/libgtk-3.0.dylib (0x4f25848) and /opt/local/lib/libgtk-3.0.dylib (0x8764948). One of the two will be used. Which one is undefined.
WARNING:
   Type initializer for class 'GdkX11DeviceXI2' (GType 'GDK-X11-DEVICE-XI2') is invalid: foreign symbol 'gdk_x11_device_xi2_get_type'
WARNING:
   Type initializer for class 'GdkX11DeviceManagerCore' (GType 'GDK-X11-DEVICE-MANAGER-CORE') is invalid: foreign symbol 'gdk_x11_device_manager_core_get_type'
WARNING:
   Type initializer for class 'GdkX11DeviceManagerXI2' (GType 'GDK-X11-DEVICE-MANAGER-XI2') is invalid: foreign symbol 'gdk_x11_device_manager_xi2_get_type'
WARNING:
   Type initializer for class 'GtkPlug' (GType 'GTK-PLUG') is invalid: foreign symbol 'gtk_plug_get_type'
WARNING:
   Type initializer for class 'GtkSocket' (GType 'GTK-SOCKET') is invalid: foreign symbol 'gtk_socket_get_type'
WARNING: cl-cffi-gtk: GType WebKitAuthenticationRequest is not known to GObject
WARNING:
   Declared GType name 'WebKitAuthenticationRequest' for class 'WEBKIT-AUTHENTICATION-REQUEST' does not match actual GType name 'GValue'
WARNING: cl-cffi-gtk: GType WebKitBackForwardListItem is not known to GObject
WARNING:
   Declared GType name 'WebKitBackForwardListItem' for class 'WEBKIT-BACK-FORWARD-LIST-ITEM' does not match actual GType name 'GInitiallyUnowned'
WARNING: cl-cffi-gtk: GType WebKitBackForwardList is not known to GObject
WARNING:
   Declared GType name 'WebKitBackForwardList' for class 'WEBKIT-BACK-FORWARD-LIST' does not match actual GType name 'GClosure'
WARNING: cl-cffi-gtk: GType WebKitCookieManager is not known to GObject
WARNING:
   Declared GType name 'WebKitCookieManager' for class 'WEBKIT-COOKIE-MANAGER' does not match actual GType name 'GFile'
WARNING: cl-cffi-gtk: GType WebKitContextMenuItem is not known to GObject
WARNING:
   Declared GType name 'WebKitContextMenuItem' for class 'WEBKIT-CONTEXT-MENU-ITEM' does not match actual GType name 'GTlsCertificateFlags'
WARNING: cl-cffi-gtk: GType WebKitContextMenu is not known to GObject
WARNING:
   Declared GType name 'WebKitContextMenu' for class 'WEBKIT-CONTEXT-MENU' does not match actual GType name 'GAppInfo'
WARNING: cl-cffi-gtk: GType WebKitDownload is not known to GObject
WARNING:
   Declared GType name 'WebKitDownload' for class 'WEBKIT-DOWNLOAD' does not match actual GType name 'GAppLaunchContext'
WARNING: cl-cffi-gtk: GType WebKitFaviconDatabase is not known to GObject
WARNING:
   Declared GType name 'WebKitFaviconDatabase' for class 'WEBKIT-FAVICON-DATABASE' does not match actual GType name 'GIcon'
WARNING: cl-cffi-gtk: GType WebKitFileChooserRequest is not known to GObject
WARNING:
   Declared GType name 'WebKitFileChooserRequest' for class 'WEBKIT-FILE-CHOOSER-REQUEST' does not match actual GType name 'GThemedIcon'
WARNING: cl-cffi-gtk: GType WebKitFindController is not known to GObject
WARNING:
   Declared GType name 'WebKitFindController' for class 'WEBKIT-FIND-CONTROLLER' does not match actual GType name 'GEmblemedIcon'
WARNING: cl-cffi-gtk: GType WebKitFormSubmissionRequest is not known to GObject
WARNING:
   Declared GType name 'WebKitFormSubmissionRequest' for class 'WEBKIT-FORM-SUBMISSION-REQUEST' does not match actual GType name 'GEmblemOrigin'
WARNING: cl-cffi-gtk: GType WebKitHitTestResult is not known to GObject
WARNING:
   Declared GType name 'WebKitHitTestResult' for class 'WEBKIT-HIT-TEST-RESULT' does not match actual GType name 'GEmblem'
WARNING: cl-cffi-gtk: GType WebKitPermissionRequest is not known to GObject
WARNING:
   Declared GType name 'WebKitPermissionRequest' for class 'WEBKIT-PERMISSION-REQUEST' does not match actual GType name 'GAction'
WARNING: cl-cffi-gtk: GType WebKitPlugin is not known to GObject
WARNING:
   Declared GType name 'WebKitPlugin' for class 'WEBKIT-PLUGIN' does not match actual GType name 'GActionGroup'
WARNING: cl-cffi-gtk: GType WebKitPrintOperation is not known to GObject
WARNING:
   Declared GType name 'WebKitPrintOperation' for class 'WEBKIT-PRINT-OPERATION' does not match actual GType name 'GActionMap'
WARNING: cl-cffi-gtk: GType WebKitScriptWorld is not known to GObject
WARNING:
   Declared GType name 'WebKitScriptWorld' for class 'WEBKIT-SCRIPT-WORLD' does not match actual GType name 'GSimpleAction'
WARNING: cl-cffi-gtk: GType WebKitFrame is not known to GObject
WARNING:
   Declared GType name 'WebKitFrame' for class 'WEBKIT-FRAME' does not match actual GType name 'GSimpleActionGroup'
WARNING: cl-cffi-gtk: GType WebKitSecurityManager is not known to GObject
WARNING:
   Declared GType name 'WebKitSecurityManager' for class 'WEBKIT-SECURITY-MANAGER' does not match actual GType name 'GApplication'
WARNING: cl-cffi-gtk: GType WebKitSettings is not known to GObject
WARNING:
   Declared GType name 'WebKitSettings' for class 'WEBKIT-SETTINGS' does not match actual GType name 'GApplicationFlags'
WARNING: cl-cffi-gtk: GType WebKitURIRequest is not known to GObject
WARNING:
   Declared GType name 'WebKitURIRequest' for class 'WEBKIT-URI-REQUEST' does not match actual GType name 'GMenuModel'
WARNING: cl-cffi-gtk: GType WebKitPolicyDecision is not known to GObject
WARNING:
   Declared GType name 'WebKitPolicyDecision' for class 'WEBKIT-POLICY-DECISION' does not match actual GType name 'GMenu'
WARNING:
   cl-cffi-gtk: GType WebKitNavigationPolicyDecision is not known to GObject
WARNING:
   Declared GType name 'WebKitNavigationPolicyDecision' for class 'WEBKIT-NAVIGATION-POLICY-DECISION' does not match actual GType name 'GMenuItem'
WARNING: cl-cffi-gtk: GType WebKitResponsePolicyDecision is not known to GObject
WARNING:
   Declared GType name 'WebKitResponsePolicyDecision' for class 'WEBKIT-RESPONSE-POLICY-DECISION' does not match actual GType name 'GListModel'
WARNING: cl-cffi-gtk: GType WebKitUserContentManager is not known to GObject
WARNING:
   Declared GType name 'WebKitUserContentManager' for class 'WEBKIT-USER-CONTENT-MANAGER' does not match actual GType name 'GListStore'
WARNING: cl-cffi-gtk: GType WebKitNetworkProxySettings is not known to GObject
WARNING:
   Declared GType name 'WebKitNetworkProxySettings' for class 'WEBKIT-NETWORK-PROXY-SETTINGS' does not match actual GType name 'GdkColorspace'
WARNING: cl-cffi-gtk: GType WebKitWebContext is not known to GObject
WARNING:
   Declared GType name 'WebKitWebContext' for class 'WEBKIT-WEB-CONTEXT' does not match actual GType name 'GdkPixbufAlphaMode'
WARNING: cl-cffi-gtk: GType WebKitWebPage is not known to GObject
WARNING:
   Declared GType name 'WebKitWebPage' for class 'WEBKIT-WEB-PAGE' does not match actual GType name 'GdkPixbuf'
WARNING: cl-cffi-gtk: GType WebKitWebResource is not known to GObject
WARNING:
   Declared GType name 'WebKitWebResource' for class 'WEBKIT-WEB-RESOURCE' does not match actual GType name 'GLoadableIcon'
WARNING: cl-cffi-gtk: GType WebKitWebsiteDataManager is not known to GObject
WARNING:
   Declared GType name 'WebKitWebsiteDataManager' for class 'WEBKIT-WEBSITE-DATA-MANAGER' does not match actual GType name 'GdkInterpType'
WARNING: cl-cffi-gtk: GType WebKitWebInspector is not known to GObject
WARNING:
   Declared GType name 'WebKitWebInspector' for class 'WEBKIT-WEB-INSPECTOR' does not match actual GType name 'GdkPixbufRotation'
WARNING: cl-cffi-gtk: GType WebKitWebView is not known to GObject
WARNING:
   Declared GType name 'WebKitWebView' for class 'WEBKIT-WEB-VIEW' does not match actual GType name 'PangoStyle'
WARNING: cl-cffi-gtk: GType WebKitWindowProperties is not known to GObject
WARNING:
   Declared GType name 'WebKitWindowProperties' for class 'WEBKIT-WINDOW-PROPERTIES' does not match actual GType name 'PangoWeight'
WARNING: cl-cffi-gtk: GType WebKitUserMessage is not known to GObject
WARNING:
   Declared GType name 'WebKitUserMessage' for class 'WEBKIT-USER-MESSAGE' does not match actual GType name 'PangoStretch'
Nyxt version 2-pre-release-4-191-gf19270de
<ERROR> [14:37:29] Could not bind socket /Users/edlewis/.local/share/nyxt/nyxt.socket, non-socket file exists.
WARNING: cl-cffi-gtk: GType WebKitWebContext is not known to GObject
WARNING: cl-cffi-gtk: GType WebKitWebContext is not known to GObject

(sbcl:92065): GLib-GObject-CRITICAL **: 14:37:29.774: g_object_newv: assertion 'G_TYPE_IS_OBJECT (object_type)' failed

(sbcl:92065): GLib-GObject-CRITICAL **: 14:37:29.774: g_object_add_toggle_ref: assertion 'G_IS_OBJECT (object)' failed

(sbcl:92065): GLib-GObject-CRITICAL **: 14:37:29.774: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

** (process:92065): CRITICAL **: 14:37:29.774: WebKitCookieManager *webkit_web_context_get_cookie_manager(WebKitWebContext *): assertion 'WEBKIT_IS_WEB_CONTEXT(context)' failed
WARNING: cl-cffi-gtk: GType WebKitWebView is not known to GObject
WARNING: cl-cffi-gtk: GType WebKitWebView is not known to GObject

(sbcl:92065): GLib-GObject-WARNING **: 14:37:29.776: cannot retrieve class for invalid (unclassed) type '<invalid>'

(sbcl:92065): GLib-GObject-CRITICAL **: 14:37:29.776: g_object_class_find_property: assertion 'G_IS_OBJECT_CLASS (class)' failed

(sbcl:92065): GLib-GObject-CRITICAL **: 14:37:29.776: g_type_class_unref: assertion 'g_class != NULL' failed
<ERROR> [14:37:29] In #<CLOSURE (LAMBDA (&OPTIONAL NYXT::URLS)
                                  :IN
                                  MAKE-STARTUP-FUNCTION) {10069F470B}>: The
                                                                        value
                                                                          NIL
                                                                        is not
                                                                        of type
                                                                          GOBJECT::PARAM-SPEC
^CUnhandled SB-SYS:INTERACTIVE-INTERRUPT in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                    {10091C8143}>:
  Interactive interrupt at #x7FFF5A77121A.

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10091C8143}>
0: ("bogus stack frame")
1: ("foreign function: __CFRunLoopServiceMachPort")
2: ("foreign function: __CFRunLoopRun")
3: ("foreign function: CFRunLoopRunSpecific")
4: ("foreign function: RunCurrentEventLoopInMode")
5: ("foreign function: ReceiveNextEventCommon")
6: ("foreign function: _BlockUntilNextEventMatchingListInModeWithFilter")
7: ("foreign function: _DPSNextEvent")
8: ("foreign function: -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]")
9: ("foreign function: poll_func")
10: ("foreign function: g_main_context_iterate")
11: ("foreign function: g_main_loop_run")
12: ("foreign function: gtk_main")
13: (GTK::%GTK-MAIN)
14: (GTK:GTK-MAIN)
15: (START :URLS NIL)
16: ((LAMBDA NIL :IN UIOP/IMAGE:RESTORE-IMAGE))
17: (UIOP/IMAGE:CALL-WITH-FATAL-CONDITION-HANDLER #<CLOSURE (LAMBDA NIL :IN UIOP/IMAGE:RESTORE-IMAGE) {100611D22B}>)
18: ((FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
19: ((FLET "WITHOUT-INTERRUPTS-BODY-1" :IN SB-IMPL::START-LISP))
20: (SB-IMPL::START-LISP)

**Comment by jmercouris** [comment]
Can you please specify your SBCL version and how you installed the dependencies? Thank you!

**Comment by ln0a** [comment]
Thanks for the fast response. I installed all the dependencies through either homebrew or mac ports. Installing Nyxt through the dedicate port produces the same result.

My sbcl version is 2.0.11

**Comment by jmercouris** [comment]
There may be conflicts with the Homebrew libraries. That is, your Lisp may be loading things sometimes via Homebrew and somtimes via Macports leading to incompatibilities. Please try putting the following into your SBCLRC:

```
(ql:quickload :cffi)
(pushnew "/opt/local/lib" cffi:*foreign-library-directories* :test #'equal)
```

this will make Lisp priorities libraries installed in that directory (used by Macports). Make sure to compile with the SBCLRC enabled. Please let me know how it goes! Thank you!

**Comment by jmercouris** [comment]
Please note, for the above snippet to work, you must have Quicklisp installed. For directions on that, please see the Quicklisp website, or our developer README.

**Comment by ln0a** [comment]
After following your instructions and changingt he Makefile to respect the user sbclrc file I still get the sane compilation `WARNINGS` and runtime issues.

--
objc[10503]: Class GNotificationCenterDelegate is implemented in both /usr/local/lib/libgio-2.0.dylib (0xb2fda58) and /opt/local/lib/libgio-2.0.0.dylib (0xfac5498). One of the two will be used. Which one is undefined.
objc[10503]: Class GdkQuartzView is implemented in both /usr/local/Cellar/gtk+3/3.24.24/lib/libgdk-3.0.dylib (0xbfacec8) and /opt/local/lib/libgdk-3.0.dylib (0xf7ea4e8). One of the two will be used. Which one is undefined.
objc[10503]: Class GdkQuartzNSWindow is implemented in both /usr/local/Cellar/gtk+3/3.24.24/lib/libgdk-3.0.dylib (0xbfacf18) and /opt/local/lib/libgdk-3.0.dylib (0xf7ea538). One of the two will be used. Which one is undefined.
objc[10503]: Class gdkCoreCursor is implemented in both /usr/local/Cellar/gtk+3/3.24.24/lib/libgdk-3.0.dylib (0xbfacf68) and /opt/local/lib/libgdk-3.0.dylib (0xf7ea588). One of the two will be used. Which one is undefined.
objc[10503]: Class GtkQuartzStatusIcon is implemented in both /usr/local/lib/libgtk-3.0.dylib (0xbe1a6b8) and /opt/local/lib/libgtk-3.0.dylib (0xf659718). One of the two will be used. Which one is undefined.
objc[10503]: Class GtkClipboardOwner is implemented in both /usr/local/lib/libgtk-3.0.dylib (0xbe1a618) and /opt/local/lib/libgtk-3.0.dylib (0xf659768). One of the two will be used. Which one is undefined.
objc[10503]: Class GtkDragSourceOwner is implemented in both /usr/local/lib/libgtk-3.0.dylib (0xbe1a668) and /opt/local/lib/libgtk-3.0.dylib (0xf6597b8). One of the two will be used. Which one is undefined.
objc[10503]: Class ResultReceiver is implemented in both /usr/local/lib/libgtk-3.0.dylib (0xbe1a730) and /opt/local/lib/libgtk-3.0.dylib (0xf659830). One of the two will be used. Which one is undefined.
objc[10503]: Class GtkApplicationQuartzDelegate is implemented in both /usr/local/lib/libgtk-3.0.dylib (0xbe1a758) and /opt/local/lib/libgtk-3.0.dylib (0xf659858). One of the two will be used. Which one is undefined.
objc[10503]: Class GNSMenuItem is implemented in both /usr/local/lib/libgtk-3.0.dylib (0xbe1a7a8) and /opt/local/lib/libgtk-3.0.dylib (0xf6598a8). One of the two will be used. Which one is undefined.
objc[10503]: Class GNSMenu is implemented in both /usr/local/lib/libgtk-3.0.dylib (0xbe1a7d0) and /opt/local/lib/libgtk-3.0.dylib (0xf6598d0). One of the two will be used. Which one is undefined.
objc[10503]: Class FilterComboBox is implemented in both /usr/local/lib/libgtk-3.0.dylib (0xbe1a848) and /opt/local/lib/libgtk-3.0.dylib (0xf659948). One of the two will be used. Which one is undefined.
WARNING:
   Type initializer for class 'GdkX11DeviceXI2' (GType 'GDK-X11-DEVICE-XI2') is invalid: foreign symbol 'gdk_x11_device_xi2_get_type'
WARNING:
   Type initializer for class 'GdkX11DeviceManagerCore' (GType 'GDK-X11-DEVICE-MANAGER-CORE') is invalid: foreign symbol 'gdk_x11_device_manager_core_get_type'
WARNING:
   Type initializer for class 'GdkX11DeviceManagerXI2' (GType 'GDK-X11-DEVICE-MANAGER-XI2') is invalid: foreign symbol 'gdk_x11_device_manager_xi2_get_type'
WARNING:
   Type initializer for class 'GtkPlug' (GType 'GTK-PLUG') is invalid: foreign symbol 'gtk_plug_get_type'
WARNING:
   Type initializer for class 'GtkSocket' (GType 'GTK-SOCKET') is invalid: foreign symbol 'gtk_socket_get_type'
WARNING: cl-cffi-gtk: GType WebKitAuthenticationRequest is not known to GObject
WARNING:
   Declared GType name 'WebKitAuthenticationRequest' for class 'WEBKIT-AUTHENTICATION-REQUEST' does not match actual GType name 'GValue'
WARNING: cl-cffi-gtk: GType WebKitBackForwardListItem is not known to GObject
WARNING:
   Declared GType name 'WebKitBackForwardListItem' for class 'WEBKIT-BACK-FORWARD-LIST-ITEM' does not match actual GType name 'GInitiallyUnowned'
WARNING: cl-cffi-gtk: GType WebKitBackForwardList is not known to GObject
WARNING:
   Declared GType name 'WebKitBackForwardList' for class 'WEBKIT-BACK-FORWARD-LIST' does not match actual GType name 'GClosure'
WARNING: cl-cffi-gtk: GType WebKitCookieManager is not known to GObject
WARNING:
   Declared GType name 'WebKitCookieManager' for class 'WEBKIT-COOKIE-MANAGER' does not match actual GType name 'GFile'
WARNING: cl-cffi-gtk: GType WebKitContextMenuItem is not known to GObject
WARNING:
   Declared GType name 'WebKitContextMenuItem' for class 'WEBKIT-CONTEXT-MENU-ITEM' does not match actual GType name 'GTlsCertificateFlags'
WARNING: cl-cffi-gtk: GType WebKitContextMenu is not known to GObject
WARNING:
   Declared GType name 'WebKitContextMenu' for class 'WEBKIT-CONTEXT-MENU' does not match actual GType name 'GAppInfo'
WARNING: cl-cffi-gtk: GType WebKitDownload is not known to GObject
WARNING:
   Declared GType name 'WebKitDownload' for class 'WEBKIT-DOWNLOAD' does not match actual GType name 'GAppLaunchContext'
WARNING: cl-cffi-gtk: GType WebKitFaviconDatabase is not known to GObject
WARNING:
   Declared GType name 'WebKitFaviconDatabase' for class 'WEBKIT-FAVICON-DATABASE' does not match actual GType name 'GIcon'
WARNING: cl-cffi-gtk: GType WebKitFileChooserRequest is not known to GObject
WARNING:
   Declared GType name 'WebKitFileChooserRequest' for class 'WEBKIT-FILE-CHOOSER-REQUEST' does not match actual GType name 'GThemedIcon'
WARNING: cl-cffi-gtk: GType WebKitFindController is not known to GObject
WARNING:
   Declared GType name 'WebKitFindController' for class 'WEBKIT-FIND-CONTROLLER' does not match actual GType name 'GEmblemedIcon'
WARNING: cl-cffi-gtk: GType WebKitFormSubmissionRequest is not known to GObject
WARNING:
   Declared GType name 'WebKitFormSubmissionRequest' for class 'WEBKIT-FORM-SUBMISSION-REQUEST' does not match actual GType name 'GEmblemOrigin'
WARNING: cl-cffi-gtk: GType WebKitHitTestResult is not known to GObject
WARNING:
   Declared GType name 'WebKitHitTestResult' for class 'WEBKIT-HIT-TEST-RESULT' does not match actual GType name 'GEmblem'
WARNING: cl-cffi-gtk: GType WebKitPermissionRequest is not known to GObject
WARNING:
   Declared GType name 'WebKitPermissionRequest' for class 'WEBKIT-PERMISSION-REQUEST' does not match actual GType name 'GAction'
WARNING: cl-cffi-gtk: GType WebKitPlugin is not known to GObject
WARNING:
   Declared GType name 'WebKitPlugin' for class 'WEBKIT-PLUGIN' does not match actual GType name 'GActionGroup'
WARNING: cl-cffi-gtk: GType WebKitPrintOperation is not known to GObject
WARNING:
   Declared GType name 'WebKitPrintOperation' for class 'WEBKIT-PRINT-OPERATION' does not match actual GType name 'GActionMap'
WARNING: cl-cffi-gtk: GType WebKitScriptWorld is not known to GObject
WARNING:
   Declared GType name 'WebKitScriptWorld' for class 'WEBKIT-SCRIPT-WORLD' does not match actual GType name 'GSimpleAction'
WARNING: cl-cffi-gtk: GType WebKitFrame is not known to GObject
WARNING:
   Declared GType name 'WebKitFrame' for class 'WEBKIT-FRAME' does not match actual GType name 'GSimpleActionGroup'
WARNING: cl-cffi-gtk: GType WebKitSecurityManager is not known to GObject
WARNING:
   Declared GType name 'WebKitSecurityManager' for class 'WEBKIT-SECURITY-MANAGER' does not match actual GType name 'GApplication'
WARNING: cl-cffi-gtk: GType WebKitSettings is not known to GObject
WARNING:
   Declared GType name 'WebKitSettings' for class 'WEBKIT-SETTINGS' does not match actual GType name 'GApplicationFlags'
WARNING: cl-cffi-gtk: GType WebKitURIRequest is not known to GObject
WARNING:
   Declared GType name 'WebKitURIRequest' for class 'WEBKIT-URI-REQUEST' does not match actual GType name 'GMenuModel'
WARNING: cl-cffi-gtk: GType WebKitPolicyDecision is not known to GObject
WARNING:
   Declared GType name 'WebKitPolicyDecision' for class 'WEBKIT-POLICY-DECISION' does not match actual GType name 'GMenu'
WARNING:
   cl-cffi-gtk: GType WebKitNavigationPolicyDecision is not known to GObject
WARNING:
   Declared GType name 'WebKitNavigationPolicyDecision' for class 'WEBKIT-NAVIGATION-POLICY-DECISION' does not match actual GType name 'GMenuItem'
WARNING: cl-cffi-gtk: GType WebKitResponsePolicyDecision is not known to GObject
WARNING:
   Declared GType name 'WebKitResponsePolicyDecision' for class 'WEBKIT-RESPONSE-POLICY-DECISION' does not match actual GType name 'GListModel'
WARNING: cl-cffi-gtk: GType WebKitUserContentManager is not known to GObject
WARNING:
   Declared GType name 'WebKitUserContentManager' for class 'WEBKIT-USER-CONTENT-MANAGER' does not match actual GType name 'GListStore'
WARNING: cl-cffi-gtk: GType WebKitNetworkProxySettings is not known to GObject
WARNING:
   Declared GType name 'WebKitNetworkProxySettings' for class 'WEBKIT-NETWORK-PROXY-SETTINGS' does not match actual GType name 'GdkColorspace'
WARNING: cl-cffi-gtk: GType WebKitWebContext is not known to GObject
WARNING:
   Declared GType name 'WebKitWebContext' for class 'WEBKIT-WEB-CONTEXT' does not match actual GType name 'GdkPixbufAlphaMode'
WARNING: cl-cffi-gtk: GType WebKitWebPage is not known to GObject
WARNING:
   Declared GType name 'WebKitWebPage' for class 'WEBKIT-WEB-PAGE' does not match actual GType name 'GdkPixbuf'
WARNING: cl-cffi-gtk: GType WebKitWebResource is not known to GObject
WARNING:
   Declared GType name 'WebKitWebResource' for class 'WEBKIT-WEB-RESOURCE' does not match actual GType name 'GLoadableIcon'
WARNING: cl-cffi-gtk: GType WebKitWebsiteDataManager is not known to GObject
WARNING:
   Declared GType name 'WebKitWebsiteDataManager' for class 'WEBKIT-WEBSITE-DATA-MANAGER' does not match actual GType name 'GdkInterpType'
WARNING: cl-cffi-gtk: GType WebKitWebInspector is not known to GObject
WARNING:
   Declared GType name 'WebKitWebInspector' for class 'WEBKIT-WEB-INSPECTOR' does not match actual GType name 'GdkPixbufRotation'
WARNING: cl-cffi-gtk: GType WebKitWebView is not known to GObject
WARNING:
   Declared GType name 'WebKitWebView' for class 'WEBKIT-WEB-VIEW' does not match actual GType name 'PangoStyle'
WARNING: cl-cffi-gtk: GType WebKitWindowProperties is not known to GObject
WARNING:
   Declared GType name 'WebKitWindowProperties' for class 'WEBKIT-WINDOW-PROPERTIES' does not match actual GType name 'PangoWeight'
WARNING: cl-cffi-gtk: GType WebKitUserMessage is not known to GObject
WARNING:
   Declared GType name 'WebKitUserMessage' for class 'WEBKIT-USER-MESSAGE' does not match actual GType name 'PangoStretch'
Nyxt version 2-pre-release-4-191-gf19270de
<ERROR> [17:19:38] Could not bind socket /Users/edlewis/.local/share/nyxt/nyxt.socket, non-socket file exists.
WARNING: cl-cffi-gtk: GType WebKitWebContext is not known to GObject
WARNING: cl-cffi-gtk: GType WebKitWebContext is not known to GObject

(sbcl:10503): GLib-GObject-CRITICAL **: 17:19:38.708: g_object_newv: assertion 'G_TYPE_IS_OBJECT (object_type)' failed

(sbcl:10503): GLib-GObject-CRITICAL **: 17:19:38.708: g_object_add_toggle_ref: assertion 'G_IS_OBJECT (object)' failed

(sbcl:10503): GLib-GObject-CRITICAL **: 17:19:38.708: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

** (process:10503): CRITICAL **: 17:19:38.708: WebKitCookieManager *webkit_web_context_get_cookie_manager(WebKitWebContext *): assertion 'WEBKIT_IS_WEB_CONTEXT(context)' failed
WARNING: cl-cffi-gtk: GType WebKitWebView is not known to GObject
WARNING: cl-cffi-gtk: GType WebKitWebView is not known to GObject

(sbcl:10503): GLib-GObject-WARNING **: 17:19:38.717: cannot retrieve class for invalid (unclassed) type '<invalid>'

(sbcl:10503): GLib-GObject-CRITICAL **: 17:19:38.717: g_object_class_find_property: assertion 'G_IS_OBJECT_CLASS (class)' failed

(sbcl:10503): GLib-GObject-CRITICAL **: 17:19:38.717: g_type_class_unref: assertion 'g_class != NULL' failed
<ERROR> [17:19:38] In #<CLOSURE (LAMBDA (&OPTIONAL NYXT::URLS)
                                  :IN
                                  MAKE-STARTUP-FUNCTION) {1006A0C86B}>: The
                                                                        value
                                                                          NIL
                                                                        is not
                                                                        of type
                                                                          GOBJECT::PARAM-SPEC

**Comment by ln0a** [comment]
Just noticed this in the compile logs:

`if [ "sbcl" = "sbcl" ]; then sbcl --no-userinit --non-interactive --eval '(assert-version->= 1 5 0)'; fi`

I will fix this now in the Makefile and see if it compiles.

**Comment by ln0a** [comment]
Unfortunately this hasn't helped.

**Comment by jmercouris** [comment]
Hm, that is unfortunate. If I think of anything else, I will comment!

**Comment by aadcg** [comment]
Nyxt has released version 3.0.0 recently. Please open a new issue if you try to run it on MacOS.
